### PR TITLE
 Refactoring Product Details to use hooks and avoid unnecessary rerender.

### PR DESCRIPTION
**Pull request by JuanAlmeida12**
#### What is the purpose of this pull request?

As the title says.

#### How should this be manually tested?

[Access this workspace](https://blurredimage--storecomponents.myvtex.com)

#### Types of changes

* [ ] Bug fix (a non-breaking change which fixes an issue)
* [ ] New feature (a non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

**Comment by jgfidelis** [comment]
Can't  you hold a reference to selectedItem instead of calling it 999 times on `render`
Like:
```
const selectedItem = getSelectedItem()
...
replace all selectedItem() to selectedItem
```

**Review by theomoura — APPROVED** [review]

File: react/ProductDetails.js [hunk 1]
```diff
@@ -1,9 +1,9 @@
 import PropTypes from 'prop-types'
-import React, { Component } from 'react'
+import React, { useCallback } from 'react'
 import { mapObjIndexed, mergeDeepRight, path } from 'ramda'
 import { injectIntl, intlShape, FormattedMessage } from 'react-intl'
 
-import { ExtensionPoint, withRuntimeContext } from 'vtex.render-runtime'
+import { ExtensionPoint, useRuntime } from 'vtex.render-runtime'
```
Comment (GustavoMA) [inline react/ProductDetails.js:19]: You need to create the mock of `useRuntime`. The tests are failing because of that.

**Review by GustavoMA — CHANGES_REQUESTED** [review]

File: react/ProductDetails.js [hunk 2]
```diff
@@ -145,29 +101,29 @@ class ProductDetails extends Component {
...
+    skuItems.sort(compareSku)
     return skuItems
   }
 
-  get selectedItem() {
-    const items = path(['productQuery', 'product', 'items'], this.props) || []
-    if (!this.props.query.skuId) return items[0]
-    return items.find(sku => sku.itemId === this.props.query.skuId)
-  }
+  const selectedItem = useCallback(() => {
```
Comment (matheusps) [inline react/ProductDetails.js]: I think `useMemo` would be a better fit.

File: react/ProductDetails.js [hunk 3]
```diff
@@ -145,29 +101,29 @@ class ProductDetails extends Component {
...
+    const items = path(['product', 'items'], productQuery) || []
+    if (!path(['skuId'], query)) return items[0]
+    return items.find(sku => sku.itemId === query.skuId)
+  }, [query])
 
-  get commertialOffer() {
-    return path(['sellers', 0, 'commertialOffer'], this.selectedItem)
-  }
+  const commertialOffer = useCallback(() => {
```
Comment (matheusps) [inline react/ProductDetails.js]: Same

File: react/ProductDetails.js [hunk 4]
```diff
@@ -145,29 +101,29 @@ class ProductDetails extends Component {
...
+    return path(['sellers', 0, 'commertialOffer'], selectedItem())
+  }, [query])
 
-  get sellerId() {
-    return parseInt(path(['sellers', 0, 'sellerId'], this.selectedItem))
-  }
+  const sellerId = useCallback(() => {
```
Comment (matheusps) [inline react/ProductDetails.js]: Same 🤓

File: react/ProductDetails.js [hunk 5]
```diff
@@ -145,29 +101,29 @@ class ProductDetails extends Component {
...
+    return parseInt(path(['sellers', 0, 'sellerId'], selectedItem()))
+  }, [query])
 
-  getImages() {
-    const images = path(['images'], this.selectedItem)
+  const getImages = useCallback(() => {
```
Comment (matheusps) [inline react/ProductDetails.js]: Here as well

File: react/ProductDetails.js [hunk 6]
```diff
@@ -179,249 +135,224 @@ class ProductDetails extends Component {
...
-              ['sellers', '0', 'commertialOffer', 'ListPrice'],
-              this.selectedItem
-            ),
-          },
-        ],
-      large: true,
-      available: showBuyButton,
-    }
+  const buyButtonProps = {
+    skuItems: selectedItem() &&
```
Comment (matheusps) [inline react/ProductDetails.js]: The change to memo would make this look better too.

File: react/ProductDetails.js [hunk 7]
```diff
@@ -179,249 +135,224 @@ class ProductDetails extends Component {
...
-                    />
-                  </div>
+          <div className={`${productDetails.nameContainer} c-on-base mb4 dn-l`}>
+            <ExtensionPoint id="product-name" {...productNameProps} />
+          </div>
+          <div className="flex flex-wrap">
+            <div className="w-60-l w-100">
+              <div className="fr-ns w-100 h-100">
+                <div className="flex justify-center">
+                  <ExtensionPoint id="product-images" images={getImages()} />
```
Comment (matheusps) [inline react/ProductDetails.js]: Do not call functions like that on component props. Could cause undesired effects.

**Review by GustavoMA — APPROVED** [review]

**Review by matheusps — APPROVED** [review]

**Comment by victorhmp** [comment]
Closing this since it became stale.

**Pull request closed without merging**